bundles/org.openhab.binding.somfymylink/src/main/java/org/openhab/binding/somfymylink/internal/handler/SomfyMyLinkBridgeHandler.java

```diff
@@ -0,0 +1,549 @@
+                                T response = parseResponse(message.toString(), responseType);
+                                logger.debug("Got full message: {}", message);
+                                future.complete(response);
+                                return;
+                            } catch (SomfyMyLinkException e) {
+                                // trouble parsing the message, probably incomplete
+                                // if its been over 1 min waiting for a correct and full response then abort
+                                elapsedTime = (new Date()).getTime() - startTime;
+                                if (elapsedTime >= 60000) {
+                                    future.completeExceptionally(
+                                            new SomfyMyLinkException("Timeout waiting for reply from mylink"));
+                                    return;
+                                }
+                                Thread.sleep(250); // pause for 250ms
+                            }
+                        }
+                    }
+
+                    // only if we didn't already get a response give time for mylink to process
+                    Thread.sleep(CONNECTION_DELAY);
+
+                    future.complete(null);
+                } catch (SocketTimeoutException e) {
+                    logger.info("Timeout sending command to mylink: {} Message: {}", command, e.getMessage());
+                    future.completeExceptionally(new SomfyMyLinkException("Timeout sending command to mylink", e));
+                } catch (IOException e) {
+                    logger.info("Problem sending command to mylink: {} Message: {}", command, e.getMessage());
+                    future.completeExceptionally(new SomfyMyLinkException("Problem sending command to mylink", e));
+                } catch (InterruptedException e) {
+                    logger.debug("Interrupted while waiting after sending command to mylink: {} Message: {}", command,
+                            e.getMessage());
+                    future.complete(null);
+                }
+            });
+            return future;
+        } else {
+            future.complete(null);
+            return future;
+        }
+    }
+
+    private <T extends SomfyMyLinkResponseBase> T parseResponse(String message, Class<T> responseType) {
+        JsonParser parser = new JsonParser();
+        JsonObject jsonObj = parser.parse(message).getAsJsonObject();
+
+        if (jsonObj != null && jsonObj.has("error")) {
+            SomfyMyLinkErrorResponse errorResponse = gson.fromJson(jsonObj, SomfyMyLinkErrorResponse.class);
+            logger.info("Error parsing mylink response: {}", errorResponse.error.message);
+            throw new SomfyMyLinkException("Incomplete message.");
+        }
+
+        return gson.fromJson(jsonObj, responseType);
+    }
+
+    private Socket getConnection() throws UnknownHostException, IOException {
```

Comment:
If an `UnknownHostException` is thrown, shouldn't the bridge thing be set to offline?

Reply:
done